hazelcast-enterprise/Dockerfile

```diff
@@ -14,83 +16,58 @@ ARG HZ_INSTALL_NAME="hazelcast-enterprise-${HZ_VERSION}"
+COPY *.xml *.sh ${HZ_HOME}/
 
-# Download and install Hazelcast plugins (hazelcast-kubernetes and hazelcast-eureka) with dependencies
-# Use Maven Wrapper to fetch dependencies specified in mvnw/dependency-copy.xml
-RUN curl -svf -o ${HZ_HOME}/maven-wrapper.tar.gz \
-         -L https://github.com/takari/maven-wrapper/archive/maven-wrapper-0.3.0.tar.gz \
- && tar zxf maven-wrapper.tar.gz \
- && rm -fr maven-wrapper.tar.gz \
- && mv maven-wrapper* mvnw
-COPY mvnw ${HZ_HOME}/mvnw
-RUN cd mvnw \
- && chmod +x mvnw \
- && sync \
- && ./mvnw -f dependency-copy.xml \
+# Install
+RUN echo "Updating Alpine system" \
+    && apk upgrade --update-cache --available \
+    && echo "Installing new APK packages" \
+    && apk add openjdk8-jre maven bash apr openssl curl procps \
+    && echo "Installing Hazelcast" \
+    && cd "${HZ_HOME}" \
+    && curl -sf -o "${HZ_HOME}/${HZ_INSTALL_ZIP}" \
+         -L "https://download.hazelcast.com/enterprise/${HZ_INSTALL_ZIP}" \
+    && unzip "${HZ_INSTALL_ZIP}" -d "${HZ_HOME}" \
+         -x "${HZ_INSTALL_NAME}/code-samples/*" \
+         -x "${HZ_INSTALL_NAME}/demo/*" \
+         -x "${HZ_INSTALL_NAME}/docs/*" \
+   && rm -rf "${HZ_INSTALL_ZIP}" \
+   && echo "Installing JCache" \
+   && curl -sf -o "${HZ_HOME}/${CACHE_API_JAR}" \
+         -L "https://repo1.maven.org/maven2/javax/cache/cache-api/${CACHE_API_VERSION}/${CACHE_API_JAR}" \
+   && echo "Installing Hazelcast plugins" \
+   && mvn -f dependency-copy.xml \
            -Dhazelcast-kubernetes-version=${HZ_KUBE_VERSION} \
            -Dhazelcast-eureka-version=${HZ_EUREKA_VERSION} \
+           -Dnetty.version=${NETTY_VERSION} \
+           -Dnetty-tcnative.version=${NETTY_TCNATIVE_VERSION} \
            dependency:copy-dependencies \
- && cd .. \
- && rm -rf $HZ_HOME/mvnw \
- && rm -rf ~/.m2 \
- && chmod -R +r $HZ_HOME
-
-ADD hazelcast.xml $HZ_HOME/hazelcast.xml
-
-# Set Pardot ID to 'docker'
-RUN echo 'hazelcastDownloadId=docker' > hazelcast-download.properties
-
-# Runtime constants
-ENV CLASSPATH_DEFAULT "${HZ_HOME}:${HZ_INSTALL_DIR}/lib/${HZ_JAR}:${HZ_HOME}/*"
-ENV JAVA_OPTS_DEFAULT "-Djava.net.preferIPv4Stack=true -Dhazelcast.mancenter.enabled=false"
+   && echo "Maven clean-up" \
+   && apk del maven \
+   && rm -rf ~/.m2 \
+   && rm -f dependency-copy.xml \
+   && echo "Granting read permission to ${HZ_HOME}" \
+   && chmod -R +r $HZ_HOME \
+   && echo "Setting Pardot ID to 'docker'" \
+   && echo 'hazelcastDownloadId=docker' > hazelcast-download.properties \
+   && echo "Cleaning APK packages" \
+   && rm -rf /var/cache/apk/*
 
-# Runtime environment variables
-ENV MIN_HEAP_SIZE ""
-ENV MAX_HEAP_SIZE ""
-ENV HZ_LICENSE_KEY ""
-ENV MANCENTER_URL ""
-
-ENV CLASSPATH ""
-ENV JAVA_OPTS ""
-
-### Expose port
-EXPOSE 5701
+WORKDIR ${HZ_HOME}
 
 # Start Hazelcast server
-CMD ["bash", "-c", "set -euo pipefail \
-      && eval JAVA_OPTS=\\\"${JAVA_OPTS}\\\" \
```

Comment:
Thanks for pointing this out, Rafal.
In general I don't like usage of `eval` because of its security implications. But here (in docker container) attacker can change `CMD` or `ENTRYPOINT` from command line directly, so the prevention in the startup script doesn't help.
I'll add it back to the script too.